File: ch10/10.3.3/ObjC/ModalViewSample/ModalViewSample/ModalViewController.m
//
//  ModalViewController.m
//  ModalViewSample
//
//  Created by tonyguan on 2016/11/2.
//  Copyright © 2016年 关东升. All rights reserved.
//

#import "ModalViewController.h"

@interface ModalViewController ()

@end

@implementation ModalViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (IBAction)save:(id)sender {
    [self dismissViewControllerAnimated:YES completion:^{
        NSLog(@"点击Save按钮，关闭模态视图");
    }];
}

- (IBAction)cancel:(id)sender {
    [self dismissViewControllerAnimated:YES completion:^{
        NSLog(@"点击Cancel按钮，关闭模态视图");
    }];
}


@end

File: ch10/10.3.3/Swift/ModalViewSample/ModalViewSample/ModalViewController.swift
//
//  ModalViewController.swift
//  ModalViewSample
//
//  Created by tonyguan on 2016/11/2.
//  Copyright © 2016年 关东升. All rights reserved.
//

import UIKit

class ModalViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }    

    @IBAction func save(_ sender: AnyObject) {
        self.dismiss(animated: true) { () -> Void in
             print("点击Save按钮，关闭模态视图")
        }
    }

    @IBAction func cancel(_ sender: AnyObject) {
        self.dismiss(animated: true) { () -> Void in
            print("点击Cancel按钮，关闭模态视图")
        }
    }
}
